report/types/report-renderer.d.ts

```diff
@@ -17,6 +17,11 @@ declare module Renderer {
 
     /** Disable the topbar UI component */
     omitTopbar?: boolean;
+    /**
+     * Convert report anchor links to a different format.
+     * Flow report uses this to convert `#seo` to `#index=0&anchor=seo`.
+     */
+    anchorTransform?: (categoryScore: HTMLAnchorElement) => HTMLAnchorElement;
```

Comment:
in the api proposal i had a slightly diff pattern:

https://github.com/GoogleChrome/lighthouse/blob/399cd3cb6b04446e53508f64d0e02cdacb429979/report/types/report-renderer.d.ts#L23-L24

i think we could use the same thing...

```ts
onPageAnchorRendered?: (el: HTMLAnchorElement) => void
```

just a callback that gets the el. doesn't try to replace it or anything. i think the general hash+onclick stuff you do still works just fine.

(naming wise.. thinking this is relevant to any "in-page" hash anchors. right now this gaugewrapper is the only one, but hypothetically if we had others, it'd all be called on them.